src/DotLiquid/Hash.cs

```diff
@@ -207,21 +220,29 @@ IEnumerator IEnumerable.GetEnumerator()
             return _nestedDictionary.GetEnumerator();
         }
 
+        /// <inheritdoc/>
         public void Add(KeyValuePair<string, object> item)
         {
             ((IDictionary<string, object>)_nestedDictionary).Add(item);
         }
 
+        /// <summary>
+        /// Determines whether the <see cref="DotLiquid.Hash">Hash</see> contains the specified key.
+        /// </summary>
+        /// <param name="key">The key to locate in the <see cref="DotLiquid.Hash">Hash</see></param>
+        /// <returns>true if the <see cref="DotLiquid.Hash">Hash</see> contains an element with the specified key or a default value; otherwise, false.</returns>
         public virtual bool Contains(object key)
         {
-            return ((IDictionary)_nestedDictionary).Contains(key);
+            return _lambda != null || _defaultValue != null || ((IDictionary) _nestedDictionary).Contains(key);

```

Comment:
nit: no space character after casting
```suggestion
            return _lambda != null || _defaultValue != null || ((IDictionary)_nestedDictionary).Contains(key);
```